### PR TITLE
Bump sqlite3

### PR DESCRIPTION
**Pull request by jessfraz**
This removes the dlopen error when compiling statically

I also included @kolyshkin  commit from https://github.com/docker/docker/pull/15387 to show we can not ignore unresoved symbols and get none now :)

### CODE REVIEW

**Comment by jfrazelle** [comment]
ping @tianon @LK4D4 @aaronlehmann

**Comment by tianon** [comment]
Nice, LGTM :+1:

**Comment by calavera** [comment]
@jfrazelle I think `vendor/src/github.com/docker/libnetwork/osl/test_linux.go` was added by mistake. Can you remove it?

File: hack/vendor.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ clone git github.com/go-check/check 64131543e7896d5bcc6bd5a76287eb75ea96c673
 clone git github.com/gorilla/context 14f550f51a
 clone git github.com/gorilla/mux e444e69cbd
 clone git github.com/kr/pty 5cf931ef8f
-clone git github.com/mattn/go-sqlite3 b4142c444a8941d0d92b0b7103a24df9cd815e42
+clone git github.com/mattn/go-sqlite3 3b3f1d01b2696af5501697c35629048c227586ab
```
Comment (tianon) [inline hack/vendor.sh]: Mind switching this to the updated version tag @mattn created for us (`v1.1.0`; https://github.com/mattn/go-sqlite3/pull/235#issuecomment-138137516)? :heart: :heart_eyes:

**Comment by calavera** [comment]
FWIW, I just opened https://github.com/docker/libnetwork/pull/499 so we don't vendor test specific functions anymore.

**Comment by jfrazelle** [comment]
cool ya because the vendoring script was adding that file

**Comment by jfrazelle** [comment]
updated

**Comment by tianon** [comment]
:metal: LGTM

**Comment by LK4D4** [comment]
LGTM

**Comment by mattn** [comment]
:metal:

**Pull request merged**